components/usage/pkg/db/cost_center_test.go

```diff
@@ -30,28 +33,77 @@ func TestCostCenter_WriteRead(t *testing.T) {
 	require.NoError(t, tx.Error)
 	require.Equal(t, costCenter.ID, read.ID)
 	require.Equal(t, costCenter.SpendingLimit, read.SpendingLimit)
+}
+
+func TestGetCostCenter(t *testing.T) {
+	conn := dbtest.ConnectForTests(t)
+	mnr := db.NewCostCenterManager(conn, db.DefaultSpendingLimit{
+		ForTeams: 0,
+		ForUsers: 500,
+	})
+	team := db.NewTeamAttributionID(uuid.New().String())
+	user := db.NewUserAttributionID(uuid.New().String())
+	cleanUp(t, conn, team, user)
 
+	teamCC, err := mnr.GetOrCreateCostCenter(context.Background(), team)
+	require.NoError(t, err)
+	userCC, err := mnr.GetOrCreateCostCenter(context.Background(), user)
+	require.NoError(t, err)
 	t.Cleanup(func() {
-		conn.Model(&db.CostCenter{}).Delete(costCenter)
+		conn.Model(&db.CostCenter{}).Delete(teamCC, userCC)
 	})
+	require.Equal(t, int32(0), teamCC.SpendingLimit)
+	require.Equal(t, int32(500), userCC.SpendingLimit)
 }
 
-func TestGetCostCenter(t *testing.T) {
+func TestSaveCostCenter(t *testing.T) {
 	conn := dbtest.ConnectForTests(t)
+	mnr := db.NewCostCenterManager(conn, db.DefaultSpendingLimit{
+		ForTeams: 0,
+		ForUsers: 500,
+	})
+	team := db.NewTeamAttributionID(uuid.New().String())
+	cleanUp(t, conn, team)
+	teamCC, err := mnr.GetOrCreateCostCenter(context.Background(), team)
+	t.Cleanup(func() {
+		conn.Model(&db.CostCenter{}).Delete(teamCC)
+	})
+	require.NoError(t, err)
+	require.Equal(t, int32(0), teamCC.SpendingLimit)
 
-	costCenter := &db.CostCenter{
-		ID:            db.NewTeamAttributionID(uuid.New().String()),
-		SpendingLimit: 300,
-	}
-
-	require.NoError(t, conn.Create(costCenter).Error)
+	teamCC.SpendingLimit = 2000
+	teamCC, err = mnr.SaveCostCenter(context.Background(), teamCC)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		conn.Model(&db.CostCenter{}).Delete(teamCC)
+	})
+	require.Equal(t, int32(2000), teamCC.SpendingLimit)
+}
 
-	results, err := db.GetCostCenter(context.Background(), conn, costCenter.ID)
+func TestSaveCostCenterMovedToStripe(t *testing.T) {
+	conn := dbtest.ConnectForTests(t)
+	mnr := db.NewCostCenterManager(conn, db.DefaultSpendingLimit{
+		ForTeams: 0,
+		ForUsers: 500,
+	})
+	team := db.NewTeamAttributionID(uuid.New().String())
+	cleanUp(t, conn, team)
+	teamCC, err := mnr.GetOrCreateCostCenter(context.Background(), team)
+	require.NoError(t, err)
+	require.Equal(t, int32(0), teamCC.SpendingLimit)
 
+	teamCC.BillingStrategy = db.CostCenter_Stripe
+	newTeamCC, err := mnr.SaveCostCenter(context.Background(), teamCC)
 	require.NoError(t, err)
-	require.Equal(t, costCenter.ID, results.ID)
+	require.Equal(t, db.CostCenter_Stripe, newTeamCC.BillingStrategy)
+	require.Equal(t, newTeamCC.CreationTime.Time().AddDate(0, 1, 0).Truncate(time.Second), newTeamCC.NextBillingTime.Time().Truncate(time.Second))
+}
 
+func cleanUp(t *testing.T, conn *gorm.DB, attributionIds ...db.AttributionID) {
 	t.Cleanup(func() {
```

Comment:
```suggestion
func cleanUp(t *testing.T, conn *gorm.DB, attributionIds ...db.AttributionID) {
    t.Helper()
	t.Cleanup(func() {
```
just for general practice